### PR TITLE
create new orders and sessions

### PR DESCRIPTION
**Pull request by vjumpkung**
I migrate from mongoose to typegoose.
Things that you can see from my old review beacuse I forgot to pull new develop (still nest mongoose) and I develop it and later it get a BIG conflict so I will make this branch instead.

Sorry for not double check develop branch before coding.

### CODE REVIEW

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: I think the type is `string` because when you response to frontend id is string (from requirement)

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: I think the type is `string[]` because when you response to frontend id is string (from requirement)

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: I think the type is `string` because when you response to frontend id is string (from requirement)

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: Type should be `() => [orders]`

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: class name should be PascalCase

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Use `HttpStatus.CREATED`

File: src/sessions/sessions.service.ts
Comment (ishiasaka) [inline src/sessions/sessions.service.ts]: Please `.exec()`

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: Please await for this

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/orders/orders.service.ts
Comment (ishiasaka) [reply src/orders/orders.service.ts]: `.exec()` this too

Comment (vjumpkung) [reply src/orders/orders.service.ts]: image

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: `createOrderDto` or simply `dto` or use another name

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: haven't solved it yet.

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: Addons is optional

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: `additional_info` is optional

File: src/orders/dto/order.create.dto.ts
Comment (ishiasaka) [inline src/orders/dto/order.create.dto.ts]: 1 File 1 Class

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/schema/order.schema.ts
Comment (lazykern) [inline src/schema/order.schema.ts]: - Please add the swagger decorator `@ApiProperty()` to every field.
- For fields with type object id, define it as a `String` in `@ApiProperty` decorator
See [src/schema/session.schema.ts](https://github.com/meeorder/meeorder-backend/blob/develop/src/schema/session.schema.ts) for references

**Review by lazykern — CHANGES_REQUESTED** [review]

File: src/session/session.service.ts
Comment (ishiasaka) [inline src/session/session.service.ts]: Please create class type for response. Don't use `Object()`

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please specific `@ApiResponse(...)`

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Don't return any response for this method And use HttpCode 204

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Don't return any response for this method And use HttpCode 204

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Don't return any response for this method And use HttpCode 204

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Don't return any response for this method And use HttpCode 204

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/orders/orders.controller.ts
Comment (lazykern) [inline src/orders/orders.controller.ts]: Should provide a way to filter orders by session state

File: src/orders/orders.controller.ts
Comment (vjumpkung) [reply src/orders/orders.controller.ts]: It's return but it's should be 201.

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: Please write Cucumber test for this

File: src/schema/order.schema.ts
Comment (ishiasaka) [inline src/schema/order.schema.ts]: `{ref: () => SessionSchema}`
Ref: https://typegoose.github.io/typegoose/docs/guides/quick-start-guide

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.controller.ts
Comment (ishiasaka) [inline src/orders/orders.controller.ts]: Please Remove`@ApiTags(...)` in method level

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: Please `.exec()` this

File: src/orders/orders.service.ts
Comment (ishiasaka) [inline src/orders/orders.service.ts]: Try to create a type or interface of this aggregation pipeline

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/orders/orders.service.ts
Comment (vjumpkung) [reply src/orders/orders.service.ts]: cannot cucumber test

Comment (vjumpkung) [reply src/orders/orders.service.ts]: can now will add some tests later.

**Review by lazykern — APPROVED** [review]

**Comment by lazykern** [comment]
Please wait for @hirasawaau approval

**Review by ishiasaka — APPROVED** [review]
Good Job!

**Pull request merged**
